### PR TITLE
Added restrictive options to login command (CRAFT-640).

### PR DESCRIPTION
**Pull request by facundobatista**
**Note**: this PR depends on #601 and #602 .

### CODE REVIEW

File: charmcraft/commands/store/__init__.py [hunk 1]
```diff
@@ -104,11 +114,82 @@ def fill_parser(self, parser):
...
+                    "can only be used when credentials are exported."
+                )
+        if parsed_args.permission is not None:
+            invalid = set(parsed_args.permission) - VALID_ATTENUATIONS
+            if invalid:
+                invalid_text = ", ".join(map(repr, sorted(invalid)))
+                raise CommandError(f"Invalid permission: {invalid_text}.")
+
+        # restrictive options, mapping the names between what is used in Namespace (singular,
+        # even if it ends up being a list) and the more natural ones used in the Store layer
```
Comment (cmatsuoka) [inline charmcraft/commands/store/__init__.py:178]: I had to read this comment a few times to understand it, then it became very obvious (to the point I would consider to removing it altogether).

**Review by cmatsuoka — APPROVED** [review]
Looks good, assuming that the store protocol is implemented correctly.

File: charmcraft/commands/store/store.py [hunk 2]
```diff
@@ -54,6 +54,15 @@
 }
 POLL_DELAY = 1
 
+# default restrictions to get auth credentials
+AUTH_DEFAULT_TTL = 3600 * 30
```
[the remark this answers is not in the document]
Comment (facundobatista) [reply charmcraft/commands/store/store.py:58]: I guess it's fine, as it's what was used before (Sergio set this value initially). We can change it in the future anyway....

File: charmcraft/commands/store/__init__.py (hunk 1 above)
Comment (facundobatista) [reply charmcraft/commands/store/__init__.py:178]: I like it, at some point explains why a "translation" is needed, we can improve it later.

**Pull request merged**